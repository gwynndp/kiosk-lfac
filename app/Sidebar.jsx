import React from "react";
import styled from "styled-components";
import { Link } from "react-router-dom";
import Logos from "./Logos";

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: flex-end;
  padding: 10px 0 10px 30px;
  min-width: 250px;
  max-width: 250px;
`;

const StyledLink = styled(Link)`
  background-color: blue;
  color: white;
  border: none;
  text-transform: uppercase;
  text-decoration: none;
  text-align: left;
  padding: 20px;
  margin: 5px 0;
  font-size: small;
  display: flex;
  align-self: stretch;
`;

export default function Sidebar() {
  return (
    <Wrapper>
      <Logos />
      <StyledLink to="/lfac">
        <span>Listening for a Change</span>
      </StyledLink>
      <StyledLink to="/schools">
        <span>Schools</span>
      </StyledLink>
      <StyledLink to="/fires">
        <span>Fire Stories</span>
      </StyledLink>
      <StyledLink to="/jackdemayo">
        <span>Remembering Jack DeMayo</span>
      </StyledLink>
      <StyledLink to="/vom">
        <span>Valley of the Moon</span>
      </StyledLink>
    </Wrapper>
  );
}
