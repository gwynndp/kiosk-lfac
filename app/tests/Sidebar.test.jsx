import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Sidebar from '../Sidebar';

test('should render a Sidebar', () => {
  const tree = renderer
    .create(
      <Router>
        <Sidebar />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

// should render the correct # of links
// should display on every page?
