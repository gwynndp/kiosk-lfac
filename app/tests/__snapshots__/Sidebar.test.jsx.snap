// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a Sidebar 1`] = `
<div
  className="sc-AykKC gVzywQ"
>
  <div
    id="logos-wrapper"
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "space-between",
        "padding": "0 10px 20px 0",
      }
    }
  >
    <img
      alt="Cloverdale City logo"
      src="images/cloverdale-city-logo.png"
      width="100%"
    />
    <img
      alt="Cloverdale High School logo"
      src="images/cloverdale-hs-logo.png"
      width="100%"
    />
    <img
      alt="Listening For a Change logo"
      src="images/lfac_logo_black.gif"
      width="100%"
    />
  </div>
  <a
    className="sc-AykKD bhXdAw"
    href="/lfac"
    onClick={[Function]}
  >
    <span>
      Listening for a Change
    </span>
  </a>
  <a
    className="sc-AykKD bhXdAw"
    href="/schools"
    onClick={[Function]}
  >
    <span>
      Schools
    </span>
  </a>
  <a
    className="sc-AykKD bhXdAw"
    href="/fires"
    onClick={[Function]}
  >
    <span>
      Fire Stories
    </span>
  </a>
  <a
    className="sc-AykKD bhXdAw"
    href="/jackdemayo"
    onClick={[Function]}
  >
    <span>
      Remembering Jack DeMayo
    </span>
  </a>
  <a
    className="sc-AykKD bhXdAw"
    href="/vom"
    onClick={[Function]}
  >
    <span>
      Valley of the Moon
    </span>
  </a>
</div>
`;
